target/linux/ramips/image/mt7621.mk

```diff
@@ -1685,6 +1685,31 @@ define Device/zio_freezio
 endef
 TARGET_DEVICES += zio_freezio
 
+define Device/zyxel_keenetic-giga-iii
+  $(Device/dsa-migration)
+  KERNEL_SIZE := 4096k  # kernel partition size
+  IMAGE_SIZE := 50m
+  BLOCKSIZE := 128k     # NAND erase size (block size)
+  PAGESIZE := 2048      # NAND page size
+  DEVICE_VENDOR := ZyXEL
+  DEVICE_MODEL := Keenetic Giga III
+  SUPPORTED_DEVICES += kng_re
```

Comment:
This line should be dropped, unless it's needed for factory flashing.

Reply:
This variable `PAGESIZE` must be specified as well as the variable `UBINIZE_OPTS := -E 5`.
See: https://github.com/openwrt/openwrt/blob/ac03e246351fbfa98e98f0319cce3a8663f836d3/include/image-commands.mk#L101-L114
See ubinize: https://github.com/sigma-star/mtd-utils/blob/f0b32c7ba5136151a961c3a43c8ff430b33d7c33/ubi-utils/ubinize.c#L47-L48
If we use a method `append-ubi`, then we must specify the variable `PAGESIZE`.

Reply:
Funny, how you discuss all of the variables here except the line I was referring to.
I was referring to the line directly before my comment, which is SUPPORTED_DEVICES ...

Reply:
You will be surprised when you enter the command:
`grep -r --include "*.mk" -n -E "SUPPORTED_DEVICES +" | wc -l`
I doubt very much that all found devices need to specify this value for factory flashing.

Reply:
grep finds several Keenetic:
```
ramips/image/mt7620.mk:1329:  SUPPORTED_DEVICES += kn_rc
ramips/image/mt7620.mk:1342:  SUPPORTED_DEVICES += kn_rf
ramips/image/mt7620.mk:1356:  SUPPORTED_DEVICES += kng_rc
```

Reply:
Have you ever considered the possibility that I know what I'm speaking of and instead of you might actually know what SUPPORTED_DEVICES _is_ good for?

Reply:
Я прежде всего доверяю написанному коду и скриптам.
The «kng_re» value is stored in the bootloader configs ("U-Config" partition).
```
servicehost=ndss.11.zyxel.ndmsystems.com
servicepass=6Wg_____480
sernumb=S150_____22
ndmhwid=kng_re
servicetag=500______776
wlanssid=Keenetic-8204
wlankey=S3i___HZ
wlanwps=658___88
country=RU
```

Reply:
Interesting. Now, would you please remove the SUPPORTED_DEVICES line?

SUPPORTED_DEVICES is for upgrades between different OpenWrt firmwares only and since this is the first OpenWrt firmware (unless OEM firmware is, which is why I asked the initial question), we don't need a custom second value here.

Reply:
OK. I will delete it. Looking at the code, I realized that it has a double meaning (not only about what you said).